llvm/include/llvm/Object/DXContainer.h

```diff
@@ -117,6 +121,49 @@ template <typename T> struct ViewArray {
 
 namespace DirectX {
 
+struct RootParameter {
+  dxbc::RootParameterHeader Header;
+  union {
+    dxbc::RootConstants Constants;
+  };
+
+  RootParameter() = default;
+};
+
+struct RootParameterView {
+  const dxbc::RootParameterHeader &Header;
+  StringRef ParamData;
+  RootParameterView(const dxbc::RootParameterHeader &H, StringRef P)
+      : Header(H), ParamData(P) {}
+
+  template <typename T> Expected<T> readParameter() {
+    T Struct;
+    if (sizeof(T) > ParamData.size())
```

Comment:
```suggestion
    if (sizeof(T) != ParamData.size())
```

I _think_ that this code should really expect that ParamData is exactly the right size.  Presumably if it were larger we'd be hiding bugs?